pkg/cloudprovider/providers/azure/azure_client.go

```diff
@@ -33,6 +34,15 @@ import (
 	"k8s.io/client-go/util/flowcontrol"
 )
 
+// Creates an error for rate limiting errors
+func createArmRateLimitErr(isWrite bool, opName string) error {
```

Comment:
Given that the purpose of this is to return this error in a channel, perhaps having this signature being:

Also nit, golang style says Arm should be ARM.

`func createARMRateLimitError(isWrite bool, opName string) (err, chan error) {`

Reply:
ah, I actually see that you use this elsewhere, perhaps two functions:

```
func createARMRateLimitError(...) error {
```

and 

```
func createARMRateLimitErrChannel() chan error {
```